tests/testlib/s2n_connection_test_utils.c

```diff
@@ -86,26 +88,10 @@ int s2n_connection_set_io_stuffers(struct s2n_stuffer *input, struct s2n_stuffer
     return 0;
 }
 
-void s2n_print_connection(struct s2n_connection *conn, const char *marker)
+void s2n_print_bytearray(uint8_t *buf, uint32_t len)
```

Comment:
Consider making this take a `FILE*`, and then 
1. You can control where the output goes
2. You can print it to a https://www.gnu.org/software/libc/manual/html_node/String-Streams.html and then check that the right thing was written in a unit test.